### PR TITLE
Avoid creating a ConcurrentMap for nongeneric instantiated witness tables

### PR DESCRIPTION
**Pull request by jckarter**

### CODE REVIEW

File: stdlib/public/runtime/Metadata.cpp [hunk 1]
```diff
@@ -4617,6 +4612,56 @@ WitnessTableCacheEntry::allocate(
...
+  void **fullTable = reinterpret_cast<void**>(this + 1);
+
+  // Zero out the witness table.
+  memset(fullTable, 0, getWitnessTableSize(conformance));
+
+  // Instantiate the table.
+  return instantiateWitnessTable(Type, Conformance, instantiationArgs, fullTable);
+}
+
+/// Instantiate the witness table for a "generic" conformance that only has
```
Comment (slavapestov) [inline stdlib/public/runtime/Metadata.cpp:4631]: This is what we call a non-dependent conformance in IRGen

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by airspeedswift** [comment]
@swift-ci please test macOS

**Pull request merged**